### PR TITLE
Added missing metadata to scheduled charge

### PR DESCRIPTION
**Pull request by AnasNaouchi**
Adding metadata support for creating scheduled charges (Missing parameter). 

* Adds param
* Updates two mocks
* Adds test
* Adds live test

### CODE REVIEW

File: src/test/java/co/omise/live/LiveScheduleRequestTest.java [hunk 1]
```diff
@@ -24,7 +26,7 @@ public void setup() throws Exception {
     }
 
     @Test
-    @Ignore("only hit the network when we need to.")
+    //@Ignore("only hit the network when we need to.")
```
Comment (nuxzero) [inline src/test/java/co/omise/live/LiveScheduleRequestTest.java:29]: Live tests supposed to be run on local only (at least for now).

**Comment by nuxzero** [comment]
@AnasNaouchi Could you add unit tests to `request` directory?

File: src/test/java/co/omise/live/LiveScheduleRequestTest.java (hunk 1 above)
Comment (AnasNaouchi) [reply src/test/java/co/omise/live/LiveScheduleRequestTest.java]: Yeah forget to uncomment.  Fixed now

**Comment by AnasNaouchi** [comment]
> @AnasNaouchi Could you add unit tests to `request` directory?

It necessary at this stage as we need this before 19th ?

**Comment by nuxzero** [comment]
Why not?

**Comment by danfowler** [comment]
> Is It necessary at this stage as we need this before 19th?

@AnasNaouchi yes, please add unit tests

**Review by nuxzero — APPROVED** [review]

File: src/test/java/co/omise/live/LiveScheduleRequestTest.java
[the remark this answers is not in the document]
Comment (AnasNaouchi) [reply src/test/java/co/omise/live/LiveScheduleRequestTest.java:1]: yes

File: file.ser
[the remark this answers is not in the document]
Comment (AnasNaouchi) [reply file.ser:1]: It's just a file that is relavant to the device running the project like the path to some sdks locally on the machine..

[the remark this answers is not in the document]
Comment (nuxzero) [reply file.ser:1]: @AnasNaouchi Do we really need to commit this file?

[the remark this answers is not in the document]
Comment (danfowler) [reply file.ser:1]: @AnasNaouchi as discussed, let's remove

File: src/test/java/co/omise/live/LiveScheduleRequestTest.java
[the remark this answers is not in the document]
Comment (danfowler) [reply src/test/java/co/omise/live/LiveScheduleRequestTest.java:1]: @AnasNaouchi as discussed, let's add tests that don't depend on live API

File: src/test/resources/testdata/fixtures/api.omise.co/schedules-post.json [hunk 2]
```diff
@@ -20,7 +20,10 @@
     "currency":"thb",
     "description":"Monthly membership fee",
     "customer":"cust_test_55bb3hkywglfyyachha",
-    "card":"card_test_55bb3fhl3tc1wr9d51d"
+    "card":"card_test_55bb3fhl3tc1wr9d51d",
+    "metadata": {
```
Comment (danfowler) [inline src/test/resources/testdata/fixtures/api.omise.co/schedules-post.json:24]: @AnasNaouchi I've added the new mocked `metadata` here

**Review by danfowler — APPROVED** [review]

**Review by aashishgurung — APPROVED** [review]

**Review by nuxzero — APPROVED** [review]

**Pull request merged**